### PR TITLE
Misc test fixes

### PR DESCRIPTION
**Pull request by ItalyPaleAle**
1. Fixes some (not all) race conditions in tests for pkg/runtime
2. Improvements to test platform and the actorfeatures test to make testing locally (outside of K8s) easier
3. Some more logging in E2E test apps
4. Increase resource allocation to Tailscale (if used), which was crashing with OOM a few times for me

Extracted from #5648

### CODE REVIEW

**Review by berndverst — DISMISSED** [review]

File: pkg/runtime/runtime_test.go [hunk 1]
```diff
@@ -5422,7 +5423,7 @@ func TestGracefulShutdownPubSub(t *testing.T) {
 	assert.NotNil(t, rt.topicRoutes)
 	go sendSigterm(rt)
 	select {
-	case <-rt.pubsubCtx.Done():
+	case <-rt.shutdownC:
```
Comment (mukundansundar) [inline pkg/runtime/runtime_test.go:5426]: this was specifically the pubsub context as it is stopped before dapr starts waiting the grace time period ... 
This was a UT specifically to test that. This change does not test that. This change just verifies that pubsub is closed after all the shutdown is down ...

**Review by mukundansundar — CHANGES_REQUESTED** [review]

File: pkg/runtime/runtime_test.go (hunk 1 above)
Comment (ItalyPaleAle) [reply pkg/runtime/runtime_test.go:5426]: I see.

I can revert this specific change but I will need @akhilac1 to look into this then. `make test-race` was failing on `pkg/runtime` on this line due to using `pubsubCtx.Done()` which then gets set to `nil` in the test.

**Review by mukundansundar — APPROVED** [review]

**Pull request merged**